custom-api/templates/store.html

```diff
@@ -0,0 +1,279 @@
+        }
+        $.ajax({
+          url: "/api/1.0/stores/{{ store_id }}/sales",
+          type: "GET",
+          data: data,
+          dataType: "json",
+          success: function(response) {
+            callback(response.request_id);
+          },
+          error: function(xhr, textStatus, errorThrown) {
+            hideLoadingIcon();
+            if (xhr.responseJSON && xhr.responseJSON.message) {
+              var errorMessage = xhr.responseJSON.message;
+              displayError(errorMessage);
+            } else {
+              displayError("An error occurred: " + textStatus);
+            }
+            callback(null);
+          }
+        });
+      }
+
+      // Poll the status for the store information request and handle results.
+      async function pollSalesStatus(requestId, resultHandler) {
+        $.ajax({
+          url: "/api/1.0/stores/{{ store_id }}/sales",
+          type: "GET",
+          data: {
+            request_id: requestId
+          },
+          dataType: "json",
+          success: function(response) {
+            // If the request is still being processed by Databricks SQL, poll again.
+            if (response.state == "PENDING" || response.state == "RUNNING") {
+              setTimeout(
+                function() { 
+                  pollSalesStatus(response.request_id, resultHandler) 
+                }, 
+                100 // 100ms polling interval.
+              )
+            // If the request is complete, process the results.
+            } else if (response.state == "SUCCEEDED") {
+              resultHandler(response.request_id, response.chunk_count);
+            } else {
+              displayError("An error occurred. Request state: " + response.state);
+            }
+          },
+          error: function(xhr, textStatus, errorThrown) {
+            hideLoadingIcon();
+            if (xhr.responseJSON && xhr.responseJSON.message) {
+              var errorMessage = xhr.responseJSON.message;
+              displayError(errorMessage);
+            } else {
+              displayError("An error occurred: " + textStatus);
+            }
+          }
+        })
+      }
+
+      // Get the external link for a given chunk.
+      async function getExternalLink(requestId, chunkIndex) {
+        const response = await $.ajax({
+          url: "/api/1.0/stores/{{ store_id }}/sales",
+          type: "GET",
+          data: {
+            request_id: requestId,
+            chunk_index: chunkIndex
+          }
+        });
+        return response.link;
+      }
+
+      // Get an external link and download the data.
+      async function downloadFromExternalLink(requestId, chunkIndex) {
+        const externalLink = await getExternalLink(requestId, chunkIndex);
+        const data = await $.ajax({
+          url: externalLink,
+          type: "GET"
+        });
+        return new Promise((resolve, reject) => {
+          resolve({chunkIndex, data});
+        });
```

Comment:
nit: since we already decorate it as `async` we don't need to return a self resolve promise, we can just return `{chunkIndex, data}`.

Reply:
It didn't work for me. Let me try again.

Reply:
ok. works now 🤷 thanks for catching!